phpstan.neon.dist

```diff
@@ -0,0 +1,17 @@
+parameters:
+    level: 3
+    paths:
+        - lib
+    ignoreErrors:
+        # Making classes final as suggested would be a BC-break
+        -
+            message: '~Unsafe usage of new static\(\)\.~'
+            paths:
+                - 'lib/Doctrine/Common/Collections/ArrayCollection.php'
+                - 'lib/Doctrine/Common/Collections/Criteria.php'
+        -
+            message: '~Array \(array\<TKey, T\>\) does not accept key int\.~'
```

Comment:
This is also suppressed for Psalm, so there doesn't seem to be a better way.

Reply:
Generally speaking, that `add()` method is broken looking from the type safety perspective. E.g. when you have `Collection<string,string>` and you call `add()` on it you're in fact mutating it into `Collection<string|int,string>`, invalidating assumption about types being constant.

Reply:
Indeed, that's not great…